server/src/main/java/org/elasticsearch/rest/MethodHandlers.java

```diff
@@ -22,26 +22,21 @@
     private final String path;
     private final Map<RestRequest.Method, Map<RestApiVersion, RestHandler>> methodHandlers;
 
-    MethodHandlers(String path, RestHandler handler, RestRequest.Method... methods) {
+    MethodHandlers(String path) {
         this.path = path;
-        this.methodHandlers = new HashMap<>(methods.length);
-        for (RestRequest.Method method : methods) {
-            methodHandlers.computeIfAbsent(method, k -> new HashMap<>())
-                .put(handler.compatibleWithVersion(), handler);
-        }
+        this.methodHandlers = new HashMap<>(4);
     }
 
     /**
      * Add a handler for an additional array of methods. Note that {@code MethodHandlers}
      * does not allow replacing the handler for an already existing method.
      */
-    MethodHandlers addMethods(RestHandler handler, RestRequest.Method... methods) {
-        for (RestRequest.Method method : methods) {
-            RestHandler existing = methodHandlers.computeIfAbsent(method, k -> new HashMap<>())
-                .putIfAbsent(handler.compatibleWithVersion(), handler);
-            if (existing != null) {
-                throw new IllegalArgumentException("Cannot replace existing handler for [" + path + "] for method: " + method);
-            }
+    MethodHandlers addMethod(RestRequest.Method method, RestApiVersion version, RestHandler handler) {
+        RestHandler existing = methodHandlers
+            .computeIfAbsent(method, k -> new HashMap<>())
```

Comment:
I wonder about the sizing here. For the majority of APIs, I think we would have at most two items in this map. So we would want the ceiling of 2 / 0.75, which is 3?

Reply:
I went in a slightly different direction, but yes, 100% and great catch that this was using the default initialCapacity, yikes. 9afd325cd756b318137f51804badf52cc8e962db